File: frontend/src/api/unscheduledTask.ts
import request from '../utils/request'
import type { ApiResponse, PageResult } from './types'

export interface UnscheduledTask {
  id: number
  batchId: number
  batchNo?: string
  taskId: number
  courseId?: number
  teacherId?: number
  classId?: number
  courseName?: string
  teacherName?: string
  className?: string
  requiredSlots: number
  scheduledSlots: number
  remainingSlots: number
  reasonType?: string
  reasonMessage?: string
  createTime: string
}

export function getUnscheduledTaskList(params: {
  batchId?: number
  courseName?: string
  teacherName?: string
  className?: string
  reasonType?: string
  page?: number
  size?: number
}) {
  return request.get<ApiResponse<PageResult<UnscheduledTask>>>('/unscheduled-tasks', { params }).then((r) => r.data.data)
}

export function clearUnscheduledTasks(batchId?: number) {
  const params: Record<string, string> = {}
  if (batchId != null) params.batchId = String(batchId)
  return request.delete('/unscheduled-tasks', { params })
}

File: backend/src/main/java/com/paike/scheduler/entity/UnscheduledTask.java
package com.paike.scheduler.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableField;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Getter;
import lombok.Setter;

import java.time.LocalDateTime;

@Getter
@Setter
@TableName("unscheduled_task")
public class UnscheduledTask {

    @TableId(type = IdType.AUTO)
    private Long id;

    private Long batchId;

    private Long semesterId;

    private Long taskId;

    private Long courseId;

    private Long teacherId;

    private Long classId;

    private Integer requiredSlots;

    private Integer scheduledSlots;

    private Integer remainingSlots;

    private String reasonType;

    private String reasonMessage;

    private LocalDateTime createTime;

    /** 关联字段 */
    @TableField(exist = false)
    private String courseName;

    @TableField(exist = false)
    private String teacherName;

    @TableField(exist = false)
    private String className;

    @TableField(exist = false)
    private String batchNo;
}
